core/TypesAndOrigins.cc

```diff
@@ -27,7 +49,12 @@ vector<ErrorLine> TypeAndOrigins::origins2Explanations(const GlobalState &gs) co
             continue;
         }
         last = o;
-        result.emplace_back(o, "");
+
+        if (originForUninitialized.exists() && o == originForUninitialized) {
+            result.emplace_back(ErrorLine::from(o, "Type may be `{}` due to uninitialized variables in:", "NilClass"));
```

Comment:
I'm not sure I'm happy with this phrasing, FWIW. It would be nice to name the specific variable that's uninitialized, but we don't have that information at this point. Also, I chose the plural `variables` here because I _think_ it's possible for more than one uninitialized variable to be part of the origins, but not sure if I like it.